common/persistence/sql/sqlplugin/postgres/plugin.go

```diff
@@ -71,6 +71,13 @@ func (d *plugin) CreateAdminDB(cfg *config.SQL) (sqlplugin.AdminDB, error) {
 // SQL database and the object can be used to perform CRUD operations on
 // the tables in the database
 func (d *plugin) createDBConnection(cfg *config.SQL) (*sqlx.DB, error) {
+	if cfg.TLS != nil && cfg.TLS.Enabled {
+		err := registerTLSConfig(cfg)
+		if err != nil {
+			return nil, fmt.Errorf("error registering TLS config: %v", err)
```

Comment:
nit: I don't see the point of wrapping the error we return in "registerTLSConfig" in a different error here.  Can you just return the same error to caller?

Reply:
agreed, removed